packages/eas-cli/src/credentials/manager/ManageIosBeta.ts

```diff
@@ -0,0 +1,125 @@
+import { findAccountByName } from '../../user/Account';
+import { ensureActorHasUsername } from '../../user/actions';
+import { Action, CredentialsManager } from '../CredentialsManager';
+import { Context } from '../context';
+//import { RemoveDistributionCertificateBeta } from '../ios/actions/RemoveDistributionCertificateBeta';
+import { AppLookupParams } from '../ios/api/GraphqlClient';
+import {
+  displayAllIosCredentials,
+  displayEmptyIosCredentials,
+} from '../ios/utils/printCredentialsBeta';
+import { PressAnyKeyToContinue } from './HelperActions';
+
+enum ActionType {
+  SetupBuildCredentials,
+  SetupBuildCredentialsFromCredentialsJson,
+  UpdateCredentialsJson,
+  UseExistingDistributionCertificate,
+  RemoveSpecificProvisioningProfile,
+  CreateDistributionCertificate,
+  UpdateDistributionCertificate,
+  RemoveDistributionCertificate,
+}
+
+export class ManageIosBeta implements Action {
+  private _currentDistributionType = null as IosDistributionType | null;
+
+  chooseDistributionType(
+    iosAppCredentials: CommonIosAppCredentialsFragment
+  ): IosDistributionType | null {
+    const iosAppBuildCredentialsArray = iosAppCredentials.iosAppBuildCredentialsArray;
+    if (iosAppBuildCredentialsArray.length === 0) {
+      return null;
+    }
+
+    // The order in which we choose the distribution type from least to most preferred
+    const typePriority = [
+      IosDistributionType.Development,
+      IosDistributionType.AdHoc,
+      IosDistributionType.Enterprise,
+      IosDistributionType.AppStore,
+    ];
+    const sortedIosAppBuildCredentialsArray = iosAppBuildCredentialsArray.sort(
+      (buildCredentialsA, buildCredentialsB) =>
+        typePriority.indexOf(buildCredentialsA.iosDistributionType) -
+        typePriority.indexOf(buildCredentialsB.iosDistributionType)
+    );
+    return sortedIosAppBuildCredentialsArray[0].iosDistributionType;
+  }
+
+  async runAsync(manager: CredentialsManager, ctx: Context): Promise<void> {
+    // TODO uncomment this before committing!
+    //manager.pushNextAction(this);
+    await ctx.bestEffortAppStoreAuthenticateAsync();
+
+    const accountName = ctx.hasProjectContext
+      ? getProjectAccountName(ctx.exp, ctx.user)
+      : ensureActorHasUsername(ctx.user);
+
+    const account = findAccountByName(ctx.user.accounts, accountName);
+    if (!account) {
+      throw new Error(`You do not have access to account: ${accountName}`);
+    }
+    if (ctx.hasProjectContext) {
+      const appLookupParams = this.getAppLookupParamsFromContext(ctx);
+      const iosAppCredentials = await ctx.newIos.getIosAppCredentialsWithBuildCredentialsAsync(
+        appLookupParams
+      );
+      if (!iosAppCredentials) {
+        displayEmptyIosCredentials(appLookupParams);
+      } else {
+        this._currentDistributionType = this.chooseDistributionType(iosAppCredentials);
+        displayAllIosCredentials(iosAppCredentials, this._currentDistributionType);
```

Comment:
I think it would be better if we display all credentials, but if you feel strongly that only one set of credentials should be displayed I suggest limiting it to only the app store and enterprise. I assume there will be a way to display all other credentials via the credentials manager actions

Reply:
Thinking about it a bit more, it makes sense to display all the credentials (app store/enterprise, adhoc, etc), instead of having an action to navigate to another configuration which is easy to miss.